### PR TITLE
chore: upgrade dependencies

### PR DESCRIPTION
**Pull request by vn7n24fzkq**
Hi! I create this PR for a few updates and improvements, as listed below:

- Updated outdated dependencies in `package.json`.
- Migrated from `.eslintrc` to `eslint.config.mjs` (to accommodate the update to ESLint v9.13.0).
- Add `dist` and `node_modules` to the ignore list in the new ESLint config file.

If this PR is accepted, I kindly request that you add the `hacktoberfest-accepted` label. Thank you!

### CODE REVIEW

File: eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import globals from 'globals';
+import path from 'node:path';
+import { fileURLToPath } from 'node:url';
+import js from '@eslint/js';
+import { FlatCompat } from '@eslint/eslintrc';
```
Comment (FlandreDaisuki) [inline eslint.config.mjs:5]: 這兩個有用到但沒宣告，預期放進 devDependencies

File: eslint.config.mjs [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import globals from 'globals';
```
Comment (FlandreDaisuki) [inline eslint.config.mjs:1]: 這一個有用到但沒宣告，預期放進 devDependencies

File: eslint.config.mjs (hunk 1 above)
Comment (vn7n24fzkq) [reply eslint.config.mjs:5]: 已補上

File: eslint.config.mjs (hunk 2 above)
Comment (vn7n24fzkq) [reply eslint.config.mjs:1]: 已補上

**Review by FlandreDaisuki — APPROVED** [review]
LGTM

**Pull request merged**